addons/base/views.py

```diff
@@ -274,6 +275,19 @@ def get_auth(auth, **kwargs):
     if not provider_settings:
         raise HTTPError(httplib.BAD_REQUEST)
 
+    path = data.get('path')
+    version = data.get('version')
+
+    if path:
+        mark_file_version_as_seen(auth.user, path, version)
+    if path and version:
+        if not node.is_contributor(auth.user):
+            file_id = path.strip('/')
+            if action == 'render':
+                update_analytics(node, file_id, version, 'view')
+            elif action == 'download':
+                update_analytics(node, file_id, version, 'download')
```

Comment:
This is getting hit even when I just view a file detail page.

I added some debug logging...

```diff
diff --git a/addons/base/views.py b/addons/base/views.py
index 7f92e497c7..0781aae42c 100644
--- a/addons/base/views.py
+++ b/addons/base/views.py
@@ -225,6 +225,8 @@ def make_auth(user):
         }
     return {}
 
+import logging
+logger = logging.getLogger(__name__)
 
 @collect_auth
 def get_auth(auth, **kwargs):
@@ -284,8 +286,10 @@ def get_auth(auth, **kwargs):
         if not node.is_contributor(auth.user):
             file_id = path.strip('/')
             if action == 'render':
+                logger.info('RENDER file_id: {}'.format(file_id))
                 update_analytics(node, file_id, version, 'view')
             elif action == 'download':
+                logger.info('DOWNLOAD file_id: {}'.format(file_id))
                 update_analytics(node, file_id, version, 'download')
 
     try:
```

And when I view a file as a non-contributor, I see both


```
[addons.base.views]  INFO: RENDER file_id: 5b981e153dfe7203a63af5aa
```


AND 

```
[addons.base.views]  INFO: DOWNLOAD file_id: 5b981e153dfe7203a63af5aa
```

@erinspace @felliott Any idea how we can prevent that spurious download count?

Reply:
Yep, was worried about the same thing! See [comment above](https://github.com/CenterForOpenScience/osf.io/pull/8496#issuecomment-420376877)

Reply:
Oh, I missed that comment.

@felliott Any idea why this is happening?

Reply:
Yes!  This is due to the nature of renderers.  MFR has three types of renderers:

1. Those that output HTML with a url pointing to WB
2. Those that download directly from WB, then output HTML containing the munged file contents.
3. Those that output HTML with a url pointing to an MFR export endpoint.

In all three cases MFR will send an auth request for a `render` action to the OSF.  I think these are the only ones that should be treated as file views.

The downloads check might need some extra logic to identify render-originating downloads.  In cases 2 & 3, MFR will include a header (`X-Cos-Mfr-Render-Request`) in the download request.  If that's present, the download can be ignored, since it's in service to a MFR render request.  Case 1 will probably involved checking the referrer and excluding those with the mfr iframe url as the referrer.

Reply:
Thanks @felliott ! @erinspace Can you look into changing the logic?